ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -677,28 +684,20 @@ std::tuple<ttnn::Tensor, ParallelConfig, ParallelConfig, bool> shard_or_reshard_
     ParallelConfig output_parallel_config =
         determine_output_parallel_config(parallel_config, compute_grid_size, out_channels, is_mm_conv);
 
-    if (needs_shard_or_reshard) {
-        auto input_shape = input_tensor.get_logical_shape();
-        if (input_shape[0] != 1 or input_shape[1] != 1) {
-            const uint32_t nhw = input_shape[0] * input_shape[1] * input_shape[2];
-            const uint32_t channels = input_shape[3];
-            ttnn::Shape new_shape({1, 1, nhw, channels});
-            input_tensor = ttnn::reshape(input_tensor, new_shape);
-            input_shape = new_shape;
-        }
+    auto input_shape = input_tensor.get_logical_shape();  // we can have unflattened (n, h, w, c) tensors here
```

Comment:
nit: use proper type instead of auto as it's not obvious from reading this line of code what is the return type.